CRM/Event/Form/Registration/Register.php

```diff
@@ -570,7 +570,7 @@ public static function formRule($fields, $files, $form) {
     }
 
     if (!$form->_allowConfirmation) {
-      $errors += CRM_Event_BAO_Participant::validateAvailableSpaces($fields + ['event_id' => $form->_eventId]);
+      $errors += CRM_Event_BAO_Participant::validateAvailableSpaces($fields + ['event_id' => $form->getEventID()]);
```

Comment:
What's the difference between `$form->getEventId()` and `$form->_values['event']['id']` (which is being used in `CRM_Event_Form_Registration_Register::checkRegistration()`?

Reply:
@jensschuppe generally we have been moving to `$form->getEventID()` where possible as that is expected to be reliable across forms and at different stages in the process (as long as it is unambigous at that point in the process). It is also tagged `@api` (as are some other similar functions like `getContactID()` and `getEventValue()`, `getContactValue()` ) which designates it as supported for extensions to interact with.

Reply:
(it's mentioned in  https://docs.civicrm.org/dev/en/latest/hooks/#pitfalls-of-hooks )